File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate( vector<int> &nums){
        //if(nums.size()<2) return false;
        unordered_set<int>noDupl;
        for ( auto i : nums){
            if(noDupl.count(i)) return true;
            noDupl.insert(i);
        }
        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int[] list = new int[k];
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int num : nums){
            map.put(num,map.getOrDefault(num,0)+1);
        }
        TreeMap<Integer,List<Integer>> freq = new TreeMap<>((a,b)->b-a);
        for(Map.Entry<Integer,Integer> entry : map.entrySet()){
            freq.computeIfAbsent(entry.getValue(),l -> new ArrayList<>()).add(entry.getKey());
        }
        int count=0;
        for(Map.Entry<Integer,List<Integer>> entry : freq.entrySet()){
            List<Integer> helpFreq = entry.getValue();
            for(int c : helpFreq){
                list[count]=c;
                count++;
                if(count==k) return list;
            }
        }
        return list;
    }
}
